src/com/ppm/integration/agilesdk/connector/jira/cloud/service/JIRAService.java

```diff
@@ -1126,7 +1141,13 @@ private JIRAIssue getIssueFromJSONObj(JSONObject obj) {
             issue.setCreationDate(fields.has("created") ? fields.getString("created") : "");
             issue.setLastUpdateDate(fields.has("updated") ? fields.getString("updated") : "");
             issue.setResolutionDate(fields.has(JIRAConstants.JIRA_FIELD_RESOLUTION_DATE) ? fields.getString(JIRAConstants.JIRA_FIELD_RESOLUTION_DATE) : "");
-            issue.setEpicKey(fields.getString(getCustomFields().epicLinkCustomField));
+            
+            if (fields.has("parent") && !fields.isNull("parent") && issue instanceof JIRAStandardIssue) {
+            	JSONObject parent = fields.getJSONObject("parent");
+            	((JIRAStandardIssue)issue).setParentIssueType(parent.getJSONObject("fields").getJSONObject("issuetype").getString("name"));
+            	((JIRAStandardIssue)issue).setParentKey(parent.getString("key"));
+            }
+            //issue.setEpicKey(fields.getString(getCustomFields().epicLinkCustomField));
```

Comment:
I will still set the epickey for the issues under epic. So that I don't need change the code in JIRACloudWorkPlanIntegration where to check if an issue is belong to an Epic. 
if (isBlank(issue.getEpicKey())) {
                            noEpicIssues.add(issue);
                        }
I kept the epickey property to make the code change less. Though we can remove it, we have to change all the places where epickey is used.